EIPS/eip-medgasprice_opcode.md

```diff
@@ -0,0 +1,93 @@
+---
+eip: <to be assigned>
+title: OPCODE 0x46 MEDGASPRICE
+author: Justice Hudson (@jchancehud)
+discussions-to: https://ethereum-magicians.org/t/medgasprice-opcode-eip/5480
+status: Draft
+type: Standards Track
+category: Core
+created: 2021-03-05
+---
+
+## Simple Summary
+
+An opcode allowing smart contracts to access information about recent gas prices.
+
+## Abstract
+
+There is an existing opcode `0x3a GASPRICE` that allows contracts to access the gas price of the current transaction. This value can be used to impose a maximum gas price on contract calls by setting a constant value, but cannot be used as a long term solution given the fluctuation of gas prices over time. The proposed opcode `0x46 MEDGASPRICE` would return the median gas price for the parent block. This would allow the implementation of upper bounds on transaction gas prices as a function of the last block median gas price.
```

Comment:
```suggestion
There is an existing opcode `GASPRICE (0x3A)` that allows contracts to access the gas price of the current transaction. This value can be used to impose a maximum gas price on contract calls by setting a constant value, but cannot be used as a long term solution given the fluctuation of gas prices over time. The proposed opcode `MEDGASPRICE (0x46)` would return the median gas price for the parent block. This would allow the implementation of upper bounds on transaction gas prices as a function of the last block median gas price.
```

Reply:
This is the EIP-1 defined way to refer to opcodes.